weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooks.ts

```diff
@@ -202,16 +203,27 @@ const useCalls = (
   filter: CallFilter,
   limit?: number,
   opts?: {skip?: boolean}
-): Loadable<CallSchema[]> => {
+): LoadableWithRefetch<CallSchema[]> => {
   const getTsClient = useGetTraceServerClientContext();
   const loadingRef = useRef(false);
   const [callRes, setCallRes] =
     useState<traceServerClient.TraceCallsQueryRes | null>(null);
   const deepFilter = useDeepMemo(filter);
+
+  const [lastFetch, setLastFetch] = useState<number | null>(null);
```

Comment:
this whole block might be cleaner if instead of relying on states to trigger hooks, we instead:
1. Create a callback called `doFetch` that does the fetch and updates the state
2. Call `doFetch` inside the current useEffect
(by this point it is a pure refactor)
3. `refetch` is then literally just the `doFetch` method (: